### ISSUE TITLE
I'm just curious how to set weights for losses

### ISSUE THREAD

**Issue by minhmanho** [avinashpaliwal/Super-SloMo#43]
As the authors reveal in their paper:
> The weights have been set empirically using a validation set as λr = 0.8, λp = 0.005, λw = 0.4, and λs = 1.

I'm just curious to know how you set the weights for losses here, is there any trick?
https://github.com/avinashpaliwal/Super-SloMo/blob/8a09e3b26d0a86359b1be57dae8b339a8ade1856/train.py#L179

**Comment by ytongW** [comment]
Did you find out how they set these weights? I am curious too!

**Comment by minhmanho** [comment]
In my opinion, these weights are set based on your training losses.
A loss represents its target, also has different difficulty to be converged. Therefore, I always ask myself a question about which loss will be useful (improve PSNR) for my final output, and how much the loss is enough? I might have to tune the weights too many times to figure out. Why don't you train the model and see what will happen :))

For example, empirically, in training, I observed that loss_smooth is easier to be converged, more than other losses, we set its weight of 1 might be okay. Meanwhile, prcpLoss is high. If we set the weight of that loss is too high, it would cause worse on final output because your network only concentrates on updating the network's params based on that loss. But if we set it as 0.005, it will improve the quality as well as the perceptual aspect.

It is too confused; therefore, they used the term "empirically" in the paper. 
I usually balance the trade-off between losses (all values are approximate), which also makes a balance in backpropagating error as well as updating weights for your target.

**Comment by ytongW** [comment]
Thanks for your kindly reply! Actually I have been training this model since yesterday, the loss decreases gradually, but the ValPSNR is always negative, can you give me a hint? Any suggests will be so helpful!

**Comment by minhmanho** [comment]
You could check:
+ Is your PSNR calculation correct?
+ (important) Run eval and check the interpolated video directly, is it good and satisfy you? If not, the model must be failed and negative PSNR is reasonable because MSE is extremely huge (more than 256*256). Negative PSNR is ... I think it's hard to appear unless there is a bug.

**Comment by ytongW** [comment]
Thank you very much!!! I will look into that.
